src/components/legend.ts

```diff
@@ -280,6 +282,7 @@ export module Components {
       entries.select("path").attr("d", (d: any, i: number, j: number) => this.symbol()(d, j)(layout.textHeight * 0.6))
                             .attr("transform", "translate(" + (layout.textHeight / 2) + "," + layout.textHeight / 2 + ")")
                             .attr("fill", (value: string) => this._colorScale.scale(value))
+                            .attr("opacity", (d: any, i: number) => this.symbolOpacity()(d, i))
```

Comment:
Can you check if the opacity based on the index of the datum actually works correctly.  We've had a bug in the past where the indices were not being piped through correctly.